pre_commit/languages/conda.py

```diff
@@ -59,16 +62,25 @@ def install_environment(
     directory = helpers.environment_dir(ENVIRONMENT_DIR, version)
 
     env_dir = prefix.path(directory)
+    env_yaml_path = prefix.path('environment.yml')
     with clean_path_on_failure(env_dir):
-        cmd_output_b(
-            'conda', 'env', 'create', '-p', env_dir, '--file',
-            'environment.yml', cwd=prefix.prefix_dir,
-        )
-        if additional_dependencies:
+        with open(env_yaml_path) as env_file:
+            env_yaml = yaml_load(env_file)
+        env_yaml['dependencies'] += additional_dependencies
+        try:
+            with NamedTemporaryFile(
+                suffix='.yml',
+                mode='w',
+                delete=False,
+            ) as tmp_env_file:
+                yaml_dump(env_yaml, stream=tmp_env_file)
+
             cmd_output_b(
-                'conda', 'install', '-p', env_dir, *additional_dependencies,
-                cwd=prefix.prefix_dir,
+                'conda', 'env', 'create', '-p', env_dir, '--file',
+                tmp_env_file.name, cwd=prefix.prefix_dir,
             )
+        finally:
+            os.remove(tmp_env_file.name)
```

Comment:
You will need to handle the case that the `NamedTemporaryFile` creation didn't work and the file / variable doesn't exist.